### ISSUE TITLE
Syncing fails to restart if Windows goes into sleep mode for a long time (e.g.: overnight)

### ISSUE THREAD

**Issue by jldinh** [ethereum/go-ethereum#2082]
Putting the computer into sleep mode and immediately waking it up is fine though.

**Comment by karalabe** [comment]
I also saw something similar, but on Linux, and not after a sleep, but rather after changing my router and adding a new one instead.

@fjl Could UPnP expiration be the cause? It would explain UDP and discovery, but it shouldn't affect the TCP connections, at least not the outbound ones. I'll try to repro my scenario with detailed logging and then see if it could be the same on windows or not.

**Comment by karalabe** [comment]
@jldinh Could you try running geth with `--vmodule=downloader=6` next time you leave it sleeping overnight and see what it outputs in the morning? Can't seem to reproduce the error on my Linux setup.

**Comment by obscuren** [comment]
@jldinh any update for us?

**Comment by jldinh** [comment]
Sorry I have been away from my node for the past week. I will give you an update later this week.

**Comment by jldinh** [comment]
OK, I managed to reproduce the issue (I forgot to mention it does not happen every time).

Here's the output:

```
I0108 00:23:44.192660   13004 blockchain.go:1230] imported 1 block(s) (0 queued
0 ignored) including 3 txs in 5.0044ms. #813963 [f53dc40e / f53dc40e]
I0108 00:23:52.934912   13004 blockchain.go:1230] imported 1 block(s) (0 queued
0 ignored) including 3 txs in 6.0049ms. #813964 [d68d022a / d68d022a]
I0108 00:23:59.353245   13004 downloader.go:228] Registering peer e7d8ba9ed62018
ad
I0108 00:24:02.759485   13004 blockchain.go:1230] imported 1 block(s) (0 queued
0 ignored) including 0 txs in 2.5026ms. #813965 [39a4eae4 / 39a4eae4]
I0108 00:24:04.613404   13004 blockchain.go:1230] imported 1 block(s) (0 queued
0 ignored) including 0 txs in 3.0027ms. #813965 [26e594b7 / 26e594b7]
I0108 00:24:19.356868   13004 downloader.go:240] Unregistering peer e7d8ba9ed620
18ad
I0108 00:24:24.661555   13004 blockchain.go:1230] imported 1 block(s) (0 queued
0 ignored) including 0 txs in 2.0024ms. #813966 [3f79ad0c / 3f79ad0c]
I0108 00:24:34.525814   13004 blockchain.go:1230] imported 1 block(s) (0 queued
0 ignored) including 2 txs in 7.0064ms. #813967 [3966fec1 / 3966fec1]
I0108 00:24:42.486602   13004 blockchain.go:1230] imported 1 block(s) (0 queued
0 ignored) including 0 txs in 3.0029ms. #813968 [92b9741a / 92b9741a]
I0108 00:24:57.822280   13004 blockchain.go:1230] imported 1 block(s) (0 queued
0 ignored) including 0 txs in 3.0026ms. #813969 [2e82202e / 2e82202e]
I0108 00:25:02.121171   13004 downloader.go:228] Registering peer e7d8ba9ed62018
ad
I0108 00:25:02.307337   13004 blockchain.go:1230] imported 1 block(s) (0 queued
0 ignored) including 0 txs in 2.0023ms. #813970 [eaa11311 / eaa11311]
I0108 00:25:25.692516   13004 blockchain.go:1230] imported 1 block(s) (0 queued
0 ignored) including 0 txs in 2.002ms. #813971 [983edaa1 / 983edaa1]
I0108 00:25:58.205468   13004 downloader.go:240] Unregistering peer e7d8ba9ed620
18ad
I0108 00:26:09.099541   13004 blockchain.go:1230] imported 1 block(s) (0 queued
0 ignored) including 13 txs in 14.0132ms. #813972 [108db943 / 108db943]
I0108 00:26:23.275349   13004 blockchain.go:1230] imported 1 block(s) (0 queued
0 ignored) including 1 txs in 5.0053ms. #813973 [fe09f7b8 / fe09f7b8]
I0108 00:26:55.704623   13004 blockchain.go:1230] imported 1 block(s) (0 queued
0 ignored) including 1 txs in 3.0029ms. #813974 [50d9b276 / 50d9b276]
I0108 09:53:20.515572   13004 downloader.go:240] Unregistering peer 37ea20c105ab
ec39
I0108 09:53:21.111854   13004 downloader.go:240] Unregistering peer bc2bfb25f2ca
95b4
I0108 09:53:21.183428   13004 downloader.go:240] Unregistering peer 624fc2892c48
e258
I0108 09:53:21.210729   13004 downloader.go:240] Unregistering peer 067cac3170ed
d6d0
I0108 09:53:21.224744   13004 downloader.go:240] Unregistering peer d0c11daea38a
adc3
I0108 09:53:21.234253   13004 downloader.go:240] Unregistering peer a979fb575495
b8d6
I0108 09:53:21.236755   13004 downloader.go:240] Unregistering peer 25155387dc8b
397a
I0108 09:53:21.243765   13004 downloader.go:240] Unregistering peer 5a89aef0ea90
d2a7
I0108 09:53:21.247769   13004 downloader.go:240] Unregistering peer d9bd2e34afa2
0e4b
I0108 09:53:21.260782   13004 downloader.go:240] Unregistering peer de471bccee3d
0422
I0108 09:53:21.261782   13004 downloader.go:240] Unregistering peer 1118980bf48b
0a36
```

Given the randomness of the issue, could it be a failure to discover new peers if all the peers known before going into sleep mode are gone when the computer wakes up?

**Comment by fjl** [comment]
This should be resolved as of geth 1.3.5.
